tests/unit-tests/lib/test_str_util.cpp

```diff
@@ -39,58 +39,317 @@ namespace test_str_util {
 
     // Tests that Foo does Xyz.
 
-    TEST_F(test_str_util, path_to_filename) {
-        std::string fname = "";
-        ASSERT_EQ(path_to_filename("/home/blah", fname), 0);
-        ASSERT_EQ(fname, "blah");
-        ASSERT_EQ(path_to_filename("hellokeith", fname), 0);
-        ASSERT_EQ(fname, "hellokeith");
-        fname = "";
-        ASSERT_EQ(path_to_filename("/home/blah/", fname), -2);
-        ASSERT_EQ(fname, "");
-        ASSERT_EQ(path_to_filename("", fname), -1);
-        ASSERT_EQ(fname, "");
+    TEST_F(test_str_util, ndays_to_string) {
+        char buf[128];
+        char *nilbuf {0};
+        EXPECT_TRUE(ndays_to_string(-1.0, 0, buf));
+        EXPECT_FALSE(ndays_to_string(0.0, -1, buf));
+        EXPECT_STREQ(buf, "");
+        EXPECT_TRUE(ndays_to_string(1.0, 0, nilbuf));
+        EXPECT_FALSE(ndays_to_string(0.0, 0, buf));
+        EXPECT_STREQ(buf, "0.00 sec ");
+        EXPECT_FALSE(ndays_to_string(1.234567890, 0, buf));
+        EXPECT_STREQ(buf, "1 days 5 hr 37 min 46.67 sec ");
+        EXPECT_FALSE(ndays_to_string(12.34567890, 1, buf));
+        EXPECT_STREQ(buf, "12 days 8 hr 17.78 min ");
+        EXPECT_FALSE(ndays_to_string(123.4567890, 2, buf));
+        EXPECT_STREQ(buf, "123 days 10.96 hr ");
+        EXPECT_FALSE(ndays_to_string(1234.567890, 3, buf));
+        EXPECT_STREQ(buf, "3 yr 138.82 days ");
+        EXPECT_FALSE(ndays_to_string(12345.67890, 4, buf));
+        EXPECT_STREQ(buf, "33.801 yr ");
+        EXPECT_FALSE(ndays_to_string(12345.67890, 5, buf));
+        EXPECT_STREQ(buf, "");
+        EXPECT_FALSE(ndays_to_string(1234.56789012345, 0, buf));
+        EXPECT_STREQ(buf, "3 yr 138 days 13 hr 37 min 45.71 sec ");
+    }
+
+    TEST_F(test_str_util, secs_to_hmsf) {
+        char buf[128];
+        //char *nilbuf {0};
+        secs_to_hmsf(0.0, buf);
+        EXPECT_STREQ(buf, "0h00m00s00");
+        secs_to_hmsf(1.0, buf);
+        EXPECT_STREQ(buf, "0h00m01s00");
+        secs_to_hmsf(1.23456789, buf);
+        EXPECT_STREQ(buf, "0h00m01s23");
+        secs_to_hmsf(12345.6789, buf);
+        EXPECT_STREQ(buf, "3h25m45s67");
+        secs_to_hmsf(123456.789, buf);
+        EXPECT_STREQ(buf, "34h17m36s78");
     }
 
     TEST_F(test_str_util, nbytes_to_string) {
         char buf[256];
         nbytes_to_string(1024, 0, buf, sizeof (buf));
-        ASSERT_STREQ(buf, "1.00 KB");
+        EXPECT_STREQ(buf, "1.00 KB");
         nbytes_to_string(1024, 1024 * 1024, buf, sizeof (buf));
-        ASSERT_STREQ(buf, "0.00/1.00 MB");
+        EXPECT_STREQ(buf, "0.00/1.00 MB");
         nbytes_to_string(512, 1024, buf, sizeof (buf));
-        ASSERT_STREQ(buf, "0.50/1.00 KB");
+        EXPECT_STREQ(buf, "0.50/1.00 KB");
         nbytes_to_string(50000000000000, 0, buf, sizeof (buf));
-        ASSERT_STREQ(buf, "45.47 TB");
+        EXPECT_STREQ(buf, "45.47 TB");
```

Comment:
Can you add these tests?  Should help cover the whole function:
nbytes_to_string(5, 0, buf, sizeof (buf));
EXPECT_STREQ(buf, "5 bytes");
nbytes_to_string(5*1024*1024, 0, buf, sizeof(buf));
EXPECT_STREQ(buf, "5.00 MB");
nbytes_to_string(15*1024*1024*1024, 0, buf, sizeof(buf));
EXPECT_STREQ(buf, "15.00 GB");
nbytes_to_string(24*1024*1024*1024*1024, 48*1024*1024*1024*1024, buf, sizeof(buf));
EXPECT_STREQ(buf, "24.00/48.00 TB");
nbytes_to_string(6*1024*1024*1024, 6*1024*1024*1024, buf, sizeof(buf));
EXPECT_STREQ(buf, "6.00/6.00 GB");
nbytes_to_string(2, 48, buf, sizeof(buf));
EXPECT_STREQ(buf, "2.00/48.00 bytes");

Reply:
done